packages/design-system/src/scalars/components/fragments/checkbox-field/checkbox-field.tsx

```diff
@@ -1,74 +1,86 @@
-import { useId } from "react";
+import React, { useId } from "react";
 import { FormLabel } from "../form-label";
 import { Checkbox, CheckboxValue } from "./checkbox";
 import { cn } from "@/scalars/lib/utils";
 import { FormMessageList } from "../form-message";
-import { FieldCommonProps } from "../../types";
+import type { FieldCommonProps } from "../../types";
 import { withFieldValidation } from "../with-field-validation";
 
-export interface CheckboxFieldProps extends FieldCommonProps<CheckboxValue> {
+type CheckboxFieldBaseProps = Omit<
+  React.ButtonHTMLAttributes<HTMLButtonElement>,
+  keyof FieldCommonProps<CheckboxValue> | "onChange"
+>;
+
+export interface CheckboxFieldProps
+  extends CheckboxFieldBaseProps,
+    FieldCommonProps<CheckboxValue> {
   onChange?: (checked: CheckboxValue) => void;
 }
 
-const CheckboxRaw: React.FC<CheckboxFieldProps> = ({
-  id: idProp,
-  name,
-  label,
-  value,
-  defaultValue,
-  disabled,
-  required,
-  description,
-  errors,
-  warnings,
-  onChange,
-  className,
-  ...props
-}) => {
-  const generatedId = useId();
-  const id = idProp ?? generatedId;
-  const hasError = !!errors?.length;
+const CheckboxRaw = React.forwardRef<HTMLButtonElement, CheckboxFieldProps>(
```

Comment:
Why are we using `HTMLButtonElement`, is not the checkbox actually a checkbox element?